UpcomingMovies/UpcomingMovies/UpcomingMoviesPage.xaml.cs

```diff
@@ -0,0 +1,203 @@
+                        return;
+                    }
+                    PopulateListView(result);
+                }
+                catch
+                {
+                    DependencyService.Get<IToast>().ShortToast("Heck, this was unexpected...");
+                }
+            });
+
+
+        }
+        void SetEvents()
+        {
+            btnSearch.Clicked += BtnSearch_Clicked;
+            lstMovies.ItemTapped += LstMovies_ItemTapped;
+            lstMovies.ItemAppearing += LstMovies_ItemAppearing;
+            sbMovieSearch.TextChanged += SbMovieSearch_TextChanged;
+        }
+        void SetConfiguration()
+        {
+            switch (Device.RuntimePlatform)
+            {
+                case Device.Android:
+                    Padding = new Thickness(0, 5, 0, 0);
+                    break;
+            }
+        }
+        void PopulateListView(UpcomingMovieModel upcomingMovieModel)
+        {
+            try
+            {
+                TotalPages = upcomingMovieModel.Total;
+                var postion = MovieListObservableCollection.Count;
+
+                upcomingMovieModel.Movies.ForEach(movie =>
+                {
+                    movie.Position = (postion++);
+                    MovieListObservableCollection.Add(movie);
+                });
+                lstMovies.ItemsSource = MovieListObservableCollection;
+            }
+            catch
+            {
+                Device.BeginInvokeOnMainThread(() =>
+                {
+                    DependencyService.Get<IToast>().ShortToast("Heck, this was unexpected...");
+                });
+            }
+        }
+        #endregion
+
+        #region Control Events
+        async void SbMovieSearch_TextChanged(object sender, TextChangedEventArgs e)
+        {
+            try
+            {
+                var isSearchTextEmpty = string.IsNullOrEmpty(sbMovieSearch.Text);
+                if (!isSearchTextEmpty)
+                {
+                    return;
+                }
+                _movieParameter.Page = 1;
+                _movieParameter.Query = string.Empty;
+                MovieListObservableCollection.Clear();
+                lstMovies.ItemsSource = MovieListObservableCollection;
+                SetPageData();
+            }
+            catch
+            {
+                Device.BeginInvokeOnMainThread(() =>
+                {
+                    DependencyService.Get<IToast>().ShortToast("Heck, this was unexpected...");
+                });
+            }
+        }
+        async void BtnSearch_Clicked(object sender, System.EventArgs e)
+        {
+            try
+            {
+                var isSearchTermReady = !string.IsNullOrEmpty(sbMovieSearch.Text) && sbMovieSearch.Text.Length > 3;
+                if (!isSearchTermReady)
+                {
+                    return;
+                }
+
+                DependencyService.Get<IToast>().ShortToast("Hang on...");
```

Comment:
Resolving services on each button click is expensive operation.

Reply:
Didn't knew about this untill now. So, what good practice should I use in this case?

Reply:
Well the simplest is to get the service once and save in the field.